index.js

```diff
@@ -54,44 +56,50 @@ function gulpRubySass (source, options) {
 		emitErr(stream, 'The sourcemap option must be true or false. See the readme for instructions on using Sass sourcemaps with gulp.');
 	}
 
-	// reassign options.sourcemap boolean to one of our two acceptable Sass arguments
 	options.sourcemap = options.sourcemap === true ? 'file' : 'none';
 
 	// create temporary directory path for the task using current working
 	// directory, source and options
-	// sass options need unix style slashes
-	var intermediateDir = slash(path.join(
+	var intermediateDir = path.join(
 		options.tempDir,
 		'gulp-ruby-sass-' + md5Hex(cwd) + md5Hex(source + JSON.stringify(options))
-	));
+	);
+
+	// Sass's single file compilation doesn't create a destination directory
+	mkdirp(intermediateDir);
+
 	var base;
-	var compileMapping;
 
+	// glob source
+	if (glob.hasMagic(source)) {
+		base = glob2base(new glob.Glob(source));
+	}
 	// directory source
-	if (path.extname(source) === '') {
-		base = path.join(cwd, source);
-		compileMapping = source + ':' + intermediateDir;
-		options.update = true;
+	else if (fs.statSync(source).isDirectory()) {
+		base = source;
 	}
-
-	// single file source
+	// file source
 	else {
-		base = path.join(cwd, path.dirname(source));
+		base = path.dirname(source);
+	}
 
-		// sass options need unix style slashes
-		var dest = slash(path.join(
-			intermediateDir,
-			gutil.replaceExtension(path.basename(source), '.css')
-		));
+	var compileMappings = glob.sync(source)
+		// remove _partials
+		.filter(function (match) {
+			return path.basename(match).indexOf('_') !== 0;
+		})
 
-		compileMapping = [ source, dest ];
+		// create source:destination arguments for Sass
+		.map(function (match) {
+			var dest = match.replace(new RegExp('^' + base), intermediateDir);
 
-		// sass's single file compilation doesn't create a destination directory, so
-		// we have to ourselves
-		mkdirp(intermediateDir);
-	}
+			if (path.extname(dest) !== '') {
+				dest = gutil.replaceExtension(dest, '.css');
+			}
 
-	// TODO: implement glob file source
+	    // sass options need unix style slashes, even if run on Windows
```

Comment:
Awesome. We require Sass 3.4, so we can definitely remove. Adding a TODO on the issue.